golem-api-grpc/proto/golem/workerexecutor/worker_executor.proto

```diff
@@ -97,6 +103,18 @@ message CreateWorkerResponse {
   }
 }
 
+message InvokeAndAwaitWorkerRequestJson {
+  golem.worker.WorkerId worker_id = 1;
+  string name = 2;
+  repeated golem.common.JsonValue input = 3;
```

Comment:
Interesting that you created a protobuf encoding for JSON, can't we just pass down the json as string to this level?

Reply:
I did think through this, and ended up deciding not to parse JSON string, and instead keep the structure and deserialize and transform instead.

However, I can also see some advantages of using `String` (probably faster?, I am not sure if parsing is more faster than dserialisation and one-to-one transformation - may be string is faster). I can see the below old version, and happy to reuse.

https://github.com/golemcloud/golem/blob/main/golem-api-grpc/proto/golem/worker/invoke_result_json.proto

Reply:
i think if we want to have structure, instead of just string, probably we should use https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.proto#L52

Reply:
I don't really see the point in modelling the JSON structure with protobuf types when in the end it's transformed back to a JSON value anyway. And probably we don't even need now to parse the input JSON in the worker service side before forwarding it to worker executor. (But even if we do, I would not worry about that much before some benchmarks/profiling shows it is the bottleneck)

Reply:
We may not need structured data here, as there can be an overhead of type informations in the payload, and therefore payload size increases. And may be json parsing at the worker-executor side might be faster than deserialisin this proto struture with extra type information - may be. This implies, I am considering @vigoo 's opinion here (just fixing my tests)

However, on the below point:

>And probably we don't even need now to parse the input JSON in the worker service side before forwarding it to worker executor. 

On this point, we already get a serde_value::Value straight away in the request handlers, and it isn't a string. So in a way we have to transform this `Value` to `String` (if we are making this a String) instead f transforming it to another structure (the one in this PR)

Also,  I am not sure if we should end up having input as `String` in an open-api spec whatsoever. So if OpenAPI spec tells the user "please pass the JSON", then what we already have is `serde_json::Value` and instead of converting it to String, we convert it to `structured` data and send it to worker-executor. So my decision to use structure here is not an accident - but just another considerable solution which I may change today to `String` solely for performance reason as mentioned above

Reply:
Still, trying to make poem pass an unparsed JSON to the request handler seems a better route to me than the protobuf encoding.